### PR TITLE
UserApp Testing

### PR DESCRIPTION
**Pull request by Muniraj132**
I’ve written the tests for the UserApp source. Please review and verify them.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208510368102325

### CODE REVIEW

File: tests/UserAppsTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        $user_id = 1;
+        $this->acting_as( $user_id );
+
+        $user_apps_services = container()->get( UserApps::class );
+        $apps = [ app_factory(), app_factory() ];
+        $user_apps_services->save_for( $user_id, $apps );
+
+        $user_apps = container()->get( UserApps::class )->raw( [ 'user_id' => $user_id ] );
+
+        $this->assertCount( 2, $user_apps );
```
Comment (incraigulous) [inline tests/UserAppsTest.php:24]: @Muniraj132, check the $slugs against the slugs generated from the app factory and make sure each slug can be found and the result.

File: tests/UserAppsTest.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        $user_id = 1;
+        $this->acting_as( $user_id );
+
+        $user_apps_services = container()->get( UserApps::class );
+        $apps = [ app_factory(), app_factory() ];
+        $user_apps_services->save_for( $user_id, $apps );
+
+        $user_apps = container()->get( UserApps::class )->for( $user_id );
+
+        $this->assertCount( 2, $user_apps );
```
Comment (incraigulous) [inline tests/UserAppsTest.php:41]: @Muniraj132, check the $slugs against the slugs generated from the app factory and make sure each slug can be found and the result.

File: tests/UserAppsTest.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+     * @test
+     */
+    public function it_saves_user_apps()
+    {
+        $user_id = 1;
+        $this->acting_as( $user_id );
+
+        $apps = [ app_factory(), app_factory() ];
+        $result = container()->get( UserApps::class )->save_for( $user_id, $apps );
+        $this->assertTrue( $result );
```
Comment (incraigulous) [inline tests/UserAppsTest.php:54]: @Muniraj132, check the $slugs against the slugs generated from the app factory and make sure each slug can be found and the result.

File: tests/UserAppsTest.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+     * @test
+     */
+    public function it_saves_user_apps_with_default_user_id()
+    {
+        $user_id = 1;
+        $this->acting_as( $user_id );
+
+        $apps = [ app_factory(), app_factory() ];
+        $result = container()->get( UserApps::class )->save( $apps );
+        $this->assertTrue( $result );
```
Comment (incraigulous) [inline tests/UserAppsTest.php:67]: @Muniraj132, fetch the user apps and check to see if they were indeed created for the user.

File: tests/UserAppsTest.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    public function it_merge_user_apps()
+    {
+        $user_id = 1;
+        $this->acting_as( $user_id );
+
+        $user_apps_services = container()->get( UserApps::class );
+        $user_apps = $user_apps_services->raw( [ 'user_id' => $user_id ] );
+
+        $new_apps = [ app_factory() ];
+        $merged_apps = $user_apps_services->merge( $user_apps, $new_apps );
```
Comment (incraigulous) [inline tests/UserAppsTest.php:82]: @Muniraj132 , I think you may be misunderstanding the merge function. It is merging an app from one source with another. 

`$user_apps_service->merge( $settings_app, $user_app );`

I would rewrite this test to adjust an app property to test if it updates.

File: tests/UserAppsTest.php [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+        // Assert that the count of merged apps is the sum of original and new apps
+        $this->assertCount( count( $user_apps ) + count( $new_apps ), $merged_apps );
+    }
```
Comment (incraigulous) [inline tests/UserAppsTest.php:86]: @Muniraj132 , You are still missing some methods from AppSource and Source.

**Review by incraigulous — CHANGES_REQUESTED** [review]

**Comment by incraigulous** [comment]
@Muniraj132 , your updates look good! Please add the rest of the methods from app source and source, and I'll review those as well.

File: tests/UserAppsTest.php (hunk 1 above)
Comment (Muniraj132) [reply tests/UserAppsTest.php:35]: @incraigulous I have completed the requested changes. I checked the $slugs against the slugs generated from the app factory to ensure that each slug can be found and returns the correct result.

File: tests/UserAppsTest.php (hunk 2 above)
Comment (Muniraj132) [reply tests/UserAppsTest.php:62]: @incraigulous I have completed the requested changes. I checked the $slugs against the slugs generated from the app factory to ensure that each slug can be found and returns the correct result.

File: tests/UserAppsTest.php (hunk 3 above)
Comment (Muniraj132) [reply tests/UserAppsTest.php:86]: @incraigulous I have completed the requested changes

**Comment by Muniraj132** [comment]
@incraigulous I have completed the requested changes. I checked the $slugs against the slugs generated from the app factory to ensure that each slug can be found and returns the correct result.

**Pull request merged**